nntrainer/src/tensor.cpp

```diff
@@ -254,59 +238,72 @@ int Tensor::subtract_i(float const &value) { return this->add_i(-value); }
 
 Tensor Tensor::subtract(float const &value) { return this->add(-value); }
 
-int Tensor::operator_i(Tensor const &m,
-                       std::function<float(float const, float const)> op) {
-  if (dim.channel() != m.dim.channel() || dim.height() != m.dim.height() ||
-      dim.width() != m.dim.width()) {
-    return ML_ERROR_INVALID_PARAMETER;
+int Tensor::operator_i(
```

Comment:
this looks much better now.
later you consider splitting this into two - `operator_i(Tensor const &m, std::function<void(const ExternalLoopInfo &e, float *, const float *)> v_func)` and a `operator_i_util(...)` for readability.

Reply:
I wasn't sure which is better. I will do that in this pr.

Reply:
Done